Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp

```diff
@@ -547,7 +547,7 @@ void WebFullScreenManager::requestExitFullScreen()
         return;
     }
 
-    if (!corePage->topDocumentHasFullscreenElement()) {
+    if (!m_element || !m_element->document().isTopDocument()) {
```

Comment:
I'm guessing there's a mismatch between `WebFullscreenManager::m_element` and `FullscreenManager::fullscreenElement()`. If I had to guess, it's the same mismatch than between `FullscreenManager::fullscreenElement()` and `FullscreenManager::m_fullscreenElement`